solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java

```diff
@@ -29,13 +30,26 @@
 import java.lang.management.MemoryMXBean;
 import java.lang.management.MemoryUsage;
 import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
 
 /**
  * FullStory: a simple servlet to produce a few prometheus metrics.
  * This servlet exists for backwards compatibility and will be removed in favor of the native prometheus-exporter.
  */
 public final class PrometheusMetricsServlet extends BaseSolrServlet {
 
+
+
+  private static Supplier<Map> getCacheStats() {
+    try {
+      return  (Supplier<Map>)SolrDispatchFilter.instance.cores.getZkController().getSolrCloudManager().getObjectCache().get("fs-shared-caches");
+    } catch (Exception e) {
+      return null;
+    }
+  }
+
```

Comment:
Why are we try/catching here instead of simply having a safe accessor that returns `null` if it's not ready or unset?

Reply:
Yeah, it can be removed